common/src/main/java/com/adaptivescale/rosetta/common/TranslationMatrix.java

```diff
@@ -182,7 +182,7 @@ private Map<Integer, List<TranslationAttributeModel>> readTranslationAttributes(
             if (translationAttributesMappedByTranslationId.containsKey(translationAttributeModel.getTranslationId())) {
                 translationAttributesMappedByTranslationId.get(translationAttributeModel.getTranslationId()).add(translationAttributeModel);
             } else {
-                translationAttributesMappedByTranslationId.put(translationAttributeModel.getTranslationId(), Arrays.asList(translationAttributeModel));
+                translationAttributesMappedByTranslationId.put(translationAttributeModel.getTranslationId(), new ArrayList<>(Arrays.asList(translationAttributeModel)));
```

Comment:
This was a bug. Using `Arrays.asList(translationAttributeModel)` creates an immutable ArrayList, where you can't add more elements